### Left-Tailed - Two Sample Z-Test (A/B Test Scenario)

#### 1 - What is it?

A `left-tailed two-sample Z-Test` checks whether the mean of **Group B** (treatment/new) is significantly **LOWER** than the mean of **Group A (control/old)**

Use this when:

    - Population standard deviations sigma1 and sigma2 are known (or assumed known)
    - Sample sizes `n1 and n2 >=30`
    - You are specifically checking for a drop, not just any difference.

`Think of it like:`
"*Did the new version hurt performance compared to the old version?*"

#### 2 - Real-Life Example: Left-Tailed Two-Sample Z-Test

We release a new UI animation. The design team fears:

    "This new animation might have REDUCED the average conversion rate."

We measure conversion(%) over two independant user groups.

- Group A (Control - old (UI))
    - mean conversion = 3.5%
    - population std1 = 1.0%
    - n1 = 400 users
- Group B (Treatment - new UI)
    - mean conversion = 3.1%
    - population std2 = 1.1%
    - n2 = 380 users

We want to test:
    
**Has conversion decreased with new UI?**

#### 3 - Hypotheses

H0 (Null): u_B = u_A
*(No decrease. New UI performs the same.)*

H1 (Alternate Left-Tailed): u_B < u_A
*(New UI leads to a decrease.)*

This is left-tailed test because we check for "less than."

#### 4 - Formula

* Difference in sample means
    - Absolute difference = x_B - x_A
* Standard error of the difference (known standard deviation case):
    - SE = SQRT((std_A^2/n_A)+(std_B^2/n_B))
* Z-statistics:
    - Z = Absolute difference / SE
        = (x_B - x_A)/SQRT((std_A^2/n_A)+(std_B^2)/n_B)
* Right-tailed p-values:
    - p = 1 - cdf(Z)
* Critical value for a = 0.05 (right-tail)
    - z_crit = stats.norm.ppf(alpha)

**Manual Step-by-step Calculation (digit-by-digit)**

In [1]:
import numpy as np
from scipy import stats

In [6]:
## Inputs:

# Group A
x_A = 3.5
std_A = 1.0
n_A = 400

# Group B
x_B = 3.1
std_B = 1.1
n_B = 380

alpha = 0.05

# Calc starts here
std_A_square = std_A ** 2
std_B_square = std_B ** 2
print(f"std square: std_A:{std_A_square:.6f} and std_B:{std_B_square:.6f}")

# 1. Difference of means:
diff = x_B - x_A
print(f"\ndifference: {diff}")

# 2. Variance contributions:
v_c_A = std_A_square/n_A
v_c_B = std_B_square/n_B
print(f"\nVariance Contribution: Group A: {v_c_A:.6f}")
print(f"Vairance Contribution: Group B: {v_c_B:.6f}")

# 3. Sum of Variance contribution
total_contributions = v_c_A + v_c_B
print(f"\nTotal Contributions: {total_contributions}")

# 4. Standard Error (SE)
SE = np.sqrt(total_contributions)
print(f"\nStandard Error(SE): {SE:.4f}")

# 5. Z-Statistics
z_stat = diff / SE
print(f"\nZ-Statistics: {z_stat:.4f}")

# 6. Left-tailed p-value
p_value = stats.norm.cdf(z_stat)
print(f"\nP-Value: {p_value:.10f}")

# 7. Critical Value
crit_value = stats.norm.ppf(alpha)
print(f"\nCritical Value: {crit_value:.4f}")

# 8. Decisions
print("\nDecision based on p_value:")
print("Reject H0: Statistically significant" if p_value < alpha else "Fail to Reject H0: NOT Statistically significatnt.")
print("\nDecision based on critical value:")
print("Reject H0" if z_stat < crit_value else "Fail to Reject H0")

std square: std_A:1.000000 and std_B:1.210000

difference: -0.3999999999999999

Variance Contribution: Group A: 0.002500
Vairance Contribution: Group B: 0.003184

Total Contributions: 0.00568421052631579

Standard Error(SE): 0.0754

Z-Statistics: -5.3055

P-Value: 0.0000000562

Critical Value: -1.6449

Decision based on p_value:
Reject H0: Statistically significant

Decision based on critical value:
Reject H0


#### 5 - Python Implementation (Step-by_Step)

In [10]:
import numpy as np
from scipy import stats

# Step1: Input
mean_A = 3.5 # control conversion (%)
sigma_A = 1.0 # control std
n_A = 400 # control sample size

mean_B = 3.1 # test conversion (%)
sigma_B = 1.1 # test std
n_B = 380 # test sample size

alpha = 0.05 # left-tailed test

# Step2: Difference & Stadard Error
diff = mean_B - mean_A
se = np.sqrt((sigma_A**2)/n_A+(sigma_B**2)/n_B)
print(f"Difference (B - A): {diff}")
print(f"Standard Error(SE): {se:.4f}")

# Step3: Z-Statistic
z_stat = diff/se
print(f"Z-Statistics: {z_stat:.4f}")

# Step4: P-Value
p_value = stats.norm.cdf(z_stat)
print(f"P-Value: {p_value:.8f}")

# Step5: Critical Value (Left Tail)
crit = stats.norm.ppf(alpha)
print(f"Critical Value: {crit:.4f}")

# Step6: Decisions
print("\n------------- Decision Based on P-Value -------------")
if p_value < alpha:
    print("Reject H0: Conversion signficantly decreased with new UI.")
else:
    print("Fail to Reject H0: No Strong evidence of a decrease.")

print("\n------------- Decision Based on P-Value -------------")
if z_stat < crit:
    print("Reject H0: Statistically significant, decrease detected")
else:
    print("Fail to Reject H0: Not enough evidence of a decrease.")

Difference (B - A): -0.3999999999999999
Standard Error(SE): 0.0754
Z-Statistics: -5.3055
P-Value: 0.00000006
Critical Value: -1.6449

------------- Decision Based on P-Value -------------
Reject H0: Conversion signficantly decreased with new UI.

------------- Decision Based on P-Value -------------
Reject H0: Statistically significant, decrease detected


#### 6 - Business Interpretation

* The treatment group's (control) conversion is 0.4 percentage points LOWER than control.
* Z = -5.31 -> This drop is **5.31 standard errors below** the control mean.
* p = 0.00000006 -> Only **0.000006% chance** this decrease happened by luck.
* Since **Z < -1.645 and p < 0.05, we reject H0.**